test/scale/scale_tester.go

```diff
@@ -178,94 +186,117 @@ scaleInTest:
 }
 
 func (t *PodSchedulingOnFailoverScaleTest) Run(client clientset.Interface, namespace *v1.Namespace, schedulerName string) {
-	statefulSets := []*resources.TestStatefulset{}
+	totalNumberOfPods := t.Replicas
+
+	nodes := nodeutil.ListAgentNodeNames(client, t.Pod.IsWindows)
+	skipper.SkipUnlessAtLeast(len(nodes), 2, "Need at least 2 agent nodes to verify the test case.")
+	midNode := len(nodes) / 2
+
+	testPod := resources.TestPod{
+		Client: client,
+	}
 
-	ginkgo.By("Initiating statefulset setup")
+	// uncordon the first half nodes
+	for _, node := range nodes[:midNode] {
+		node := node
+		framework.Logf("Uncordoning %s.", node)
+		testPod.SetNodeUnschedulable(node, false)
+	}
+
+	// cordon the second half nodes)
+	for _, node := range nodes[midNode:] {
+		node := node
+		framework.Logf("Cordoning %s.", node)
+		testPod.SetNodeUnschedulable(node, true)
+	}
 
+	// start scale out test
+	startScaleOut := time.Now()
 	tStorageClass, storageCleanup := t.Pod.CreateStorageClass(client, namespace, t.CSIDriver, t.StorageClassParameters)
 	defer storageCleanup()
+	statefulSet, cleanupStatefulSet := t.Pod.SetupStatefulset(client, namespace, t.CSIDriver, schedulerName, t.Replicas, &tStorageClass, map[string]string{"group": "azuredisk-scale-tester"})
+	defer cleanupStatefulSet(45 * time.Minute)
+
+	statefulSet.CreateWithoutWaiting()
+	ticker := time.NewTicker(time.Minute)
+	numberOfRunningPods1 := podutil.WaitForPodsWithMatchingLabelToRun(*scaleTestTimeout, ticker, totalNumberOfPods, client, namespace)
```

Comment:
You could initialize the `ticker` in the `WaitForPodsWithMatchingLabelToRun` instead of passing it to the function.